scripts/rust-libc.py

```diff
@@ -0,0 +1,699 @@
+#!/bin/env python3
+
+# HOW THIS WORKS
+#
+# This script takes mlibc header files and generates bindings to be used with rust's "libc" crate.
+# A configuration file is needed for its proper function; an example is provided alongside this
+# script. Please do note that it is used for managarm, which lives under `unix/linux_like` in the
+# "libc" crate. If your OS does not live under this directory, but e.g. under just `unix` instead,
+# you will need to adapt the configuration to fit your use.
+#
+# HOW TO USE
+#
+# > python rust-libc <path/to/your/installed/mlibc/headers> [<single-header.h>]
+#
+# By default, the script parses all header files in the directory supplied, except for when a
+# single header is provided, where it will only parse that.
+
+import argparse
+import binascii
+import clang.cindex
+import colorama
+import functools
+import io
+import itertools
+import pathlib
+import string
+import sys
+import yaml
+
+from clang.cindex import (TypeKind, CursorKind, TokenKind, Cursor)
+
+dry_run = True
+errors_emitted = 0
+
+def log_err(prefix, msg):
+	global errors_emitted
+
+	print(f"{colorama.Fore.RED}{prefix}{colorama.Style.RESET_ALL}: {msg}")
+	errors_emitted += 1
+
+def emit(msg):
+	if not dry_run:
+		print(msg)
```

Comment:
Why not just output to stdout instead of having dry_run etc? Then user can redirect where they want, and errors logged to stderr